wbooks/src/app/components/BookCard/index.tsx

```diff
@@ -1,26 +1,36 @@
 import React from 'react';
 import { View, Image } from 'react-native';
-import CustomText from '@components/CustomText';
+import { TouchableOpacity } from 'react-native-gesture-handler';
 import { Book } from '@interfaces/books';
+import { useNavigation } from '@react-navigation/native';
+import routes from '@constants/routes';
+import bookImgPlaceholder from '@assets/img_book_placeholder.png';
+import CustomText from '@components/CustomText';
 
 import styles, { oneLineText } from './styles';
-import bookImgPlaceholder from './assets/img_book_placeholder.png';
 
 interface Props extends Partial<Book> {
+  id: number;
   imageUrl: string | null;
   title: string;
   author: string;
 }
 
-function BookCard({ imageUrl, title, author }: Props) {
+function BookCard({ id, imageUrl, title, author }: Props) {
+  const navigation = useNavigation();
+
+  const onBookClicked = () => navigation.navigate(routes.BookDetail, { bookId: id });
```

Comment:
Instead of sending an `id` + `find by id`, you can pass the book item directly:
```
function BookCard({ book }: Props) {
  const { id, imageUrl, title, author } = book;
  const onBookClicked = () => navigation.navigate(routes.BookDetail, { book });
...
}
```